service/dap/server.go

```diff
@@ -1067,13 +1067,9 @@ func (s *Session) stopNoDebugProcess() {
 		// We already handled termination or there was never a process
 		return
 	}
-	if s.noDebugProcess.ProcessState != nil && s.noDebugProcess.ProcessState.Exited() {
-		s.logToConsole(proc.ErrProcessExited{Pid: s.noDebugProcess.ProcessState.Pid(), Status: s.noDebugProcess.ProcessState.ExitCode()}.Error())
-	} else {
-		// TODO(hyangah): gracefully terminate the process and its children processes.
-		s.logToConsole(fmt.Sprintf("Terminating process %d", s.noDebugProcess.Process.Pid))
-		s.noDebugProcess.Process.Kill() // Don't check error. Process killing and self-termination may race.
-	}
+	// TODO(hyangah): gracefully terminate the process and its children processes.
+	s.logToConsole(fmt.Sprintf("Terminating process %d", s.noDebugProcess.Process.Pid))
+	s.noDebugProcess.Process.Kill() // Don't check error. Process killing and self-termination may race.
```

Comment:
@hyangah Which does comment refer to? The server won't try to self-terminate until stopNoDebugProcess function returns. What exactly can cause a race?

Reply:
The program's self-termination.

The goroutine that's waiting on the program's termination with Command.Wait() may not have observed it yet.